modules/references/client/components/read-references/Reference.js

```diff
@@ -116,7 +116,7 @@ export default function Reference({ experience }) {
             <TimeAgo date={createdDate} />
           </ReferenceLink>
         </Header>
-        {!isPublicReference && (
+        {!isPublicReference && feedbackPublic === undefined && (
```

Comment:
Checking for `feedbackPublic` being undefined for this scenario feels a bit fragile, but I also don't have better suggestions. 🤔

Reply:
I don't' like it either. It feels like the `Reference` component needs some parameter from the outside like `feedbackVisible` or `feedbackObfuscated`, or instead of "feedback" use some other term that would summarize textual feedback, recommendation, and interactions.

Reply:
In general, I think the root cause of such robustness issues is that, in our `Experience` object, there is this group of keys (`recommend`, `feedbackPublic`, `interactions`) that conceptually belong together: 1) they are all created by the user directly when creating experience 2) they all represent the opinion of the user 3) they all need to be obfuscated when returning from the back-end if the experience is still private.  If there were some `feedback` key, e.g.:
```
experience = {
    _id,
    userFrom,
    userTo,
    created,
    feedback: {
         textualFeedbackPublic,
         interactions,
         recommend,
    },
    responseCreated,
    responseFeedback: {
         textualFeedbackPublic,
         interactions,
         recommend,
    }
}
```
..., the back-end wouldn't return the whole `feedback` object. Would comparing `feedback` with `undefined` make things look more robust?